tmt/steps/execute/__init__.py

```diff
@@ -562,31 +562,65 @@ def load_tmt_report_results(self, invocation: TestInvocation) -> list["tmt.Resul
             return []
 
         # Check the test result
-        self.debug(f"tmt-report-results file '{report_result_path} detected.")
+        self.debug(f"tmt-report-results file '{report_result_path}' detected.")
+
+        # The worst result outcome we can find among loaded results...
+        original_outcome: Optional[ResultOutcome] = None
+        # ... and the actual outcome we decided is the best representing
+        # the results.
+        # The original one may be left unset - malformed results file,
+        # for example, provides no usable original outcome.
+        actual_outcome: ResultOutcome
+        note: Optional[str] = None
+
+        results = tmt.utils.yaml_to_list(self.read(report_result_path))
 
-        with open(report_result_path) as result_file:
-            result_list = [line for line in result_file.readlines() if "TESTRESULT" in line]
-        if not result_list:
+        if not results:
             raise tmt.utils.ExecuteError(
                 f"Test result not found in result file '{report_result_path}'.")
-        result = result_list[0].split("=")[1].strip()
-
-        # Map the restraint result to the corresponding tmt value
-        actual_result = ResultOutcome.ERROR
-        note: Optional[str] = None
 
         try:
-            actual_result = ResultOutcome(result.lower())
-        except ValueError:
-            if result == 'SKIP':
-                actual_result = ResultOutcome.INFO
-            else:
-                note = f"invalid test result '{result}' in result file"
+            outcomes = [
+                ResultOutcome.from_spec(result.get('result'))
+                for result in results
+                ]
+
+        except tmt.utils.SpecificationError as exc:
+            actual_outcome = ResultOutcome.ERROR
+            note = exc.message
+
+        else:
+            hierarchy = [
+                ResultOutcome.SKIP,
+                ResultOutcome.PASS,
+                ResultOutcome.WARN,
+                ResultOutcome.FAIL]
+            outcome_indices = [hierarchy.index(outcome) for outcome in outcomes]
+            actual_outcome = original_outcome = hierarchy[max(outcome_indices)]
+
+        # Find a usable log - the first one matching our "interim" outcome.
+        # We cannot use the "actual" outcome, because that one may not even
+        # exist in the results file - tmt might have conjured it based on
+        # provided results, or set it to "error" because of errors. Only
+        # the "interim" is guaranteed to be found among the results.
+        test_logs = [invocation.relative_path / TEST_OUTPUT_FILENAME]
+
+        if original_outcome is not None:
```

Comment:
original_outcome is only used here, IMO it could be made simpler by dropping it since it is the same as actual_outcome for all results except ERROR where it stays on None, so you could just have `actual_outcome is not ResultOutcome.ERROR` instead

Reply:
I for one would keep it. It's clear the decision is based on the original result, which is less clear from `actual_outcome is not ResultOutcome.ERROR`, and if things crash, the original outcome would be available in `locals()` & shown in the traceback. The original code was a bit muddy, I hoped the split between "this is what we read" and "this is what we decide to propagate" would improve the readability.

Reply:
ok, I guess that makes sense, like this it's immediately understandable what each of those variables contain

to me it just seemed like there was not much use for it, but either is fine with me